lib/core/src/Cardano/Wallet.hs

```diff
@@ -3432,16 +3432,15 @@ data ErrBalanceTx
     | ErrBalanceTxSelectAssets ErrSelectAssets
     | ErrBalanceTxMaxSizeLimitExceeded
     | ErrBalanceTxExistingCollateral
+    | ErrBalanceTxConflictingNetworks
     | ErrBalanceTxAssignRedeemers ErrAssignRedeemers
-    | ErrBalanceTxNotYetSupported BalanceTxNotSupportedReason
-    | ErrBalanceTxFailedBalancing Cardano.Value
+    | ErrBalanceTxInternalError ErrBalanceTxInternalError
+    | ErrBalanceTxZeroAdaOutput
     deriving (Show, Eq)
 
--- TODO: Remove once problems are fixed.
-data BalanceTxNotSupportedReason
-    = UnderestimatedFee Coin SealedTx
-    | ZeroAdaOutput
-    | ConflictingNetworks
+data ErrBalanceTxInternalError
```

Comment:
Not 100% about this, but think it might make sense to bundle the internal, "impossible" errors separately from the rest.